core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -718,9 +718,9 @@ public PlanNode plan(Update node)
         return createMergePipeline(table, relationPlan, projectNode, rowIdSymbol, mergeRowSymbol);
     }
 
-    private PlanBuilder addCheckConstraints(List<Expression> constraints, PlanBuilder subPlanBuilder)
+    private PlanBuilder addCheckConstraints(List<Expression> constraints, PlanBuilder subPlanBuilder, RelationPlan targetTablePlan)
     {
-        PlanBuilder constraintBuilder = subPlanBuilder.appendProjections(constraints, symbolAllocator, idAllocator);
+        PlanBuilder constraintBuilder = subPlanBuilder.withScope(targetTablePlan.getScope(), targetTablePlan.getFieldMappings()).appendProjections(constraints, symbolAllocator, idAllocator);
```

Comment:
The withScope call should be done by the caller. The subPlanBuilder should be properly set up coming into this method to be able to translate the constraint expressions.